### PR TITLE
container image name consolidation

### PR DESCRIPTION
**Pull request by k15r**
**Description**

Changes proposed in this pull request:

- Propose a naming pattern for container images

**Related issue(s)**

### CODE REVIEW

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+The path to each binary is unique. Significant information contained in the path of a binary should be used as the base for the image name that contains the binary.
+
+### Components:
+
+#### Pattern:
+
+| path | image name | rule |
+| ---- | ---------- | ---- |
+| `components/<component-name>/cmd/<component-binary>` | `<component-name>-<component-binary>` | applies if multiple images can be built using the component-folder |
+| `components/<component-name>/cmd/<component-binary>` | `<component-name>` | applies if only a single images can be built using the component-folder |
```
Comment (mszostok) [inline collaboration/sig-core/proposals/image-name-consolidation.md]: just add info that `cmd` is not required and if someone has only one `main.go` on root directory then the same pattern applies :)

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+### Tests:
+
+ #### Patterns:
+ 
+| path | image name |
+| ---- | ---------- |
+| `tests/<test-type>/<test-name>` | `<test-name>-<test-type>-tests` |
+| `tests/<component-name>` | `<component-name>-tests` |
+
+#### Example
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]: Please remove the unnecessary heading.
```suggestion
See the following examples:
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+| path | image name |
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
| Path | Image name |
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+| ---- | ---------- |
+| `tests/end-to-end/upgrade` | `upgrade-end-to-end-test` |
+| `tests/knative-serving` | `knative-serving-test` |
+
+
+## Actions
+
+- Update the developer's guide to include the naming convention
+- Update all Makefiles to use the new naming convention
+- Update all helm charts to reference the new images
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
- Update all Helm charts to reference the new images.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+
+> Components and tests not listed below already adhere to the patterns described above
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
> Components and tests not listed below already adhere to the patterns described above.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+
+### Components
+Currently all images follow the proposed pattern and exclusions.
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
Currently, all images follow the proposed pattern and exclusions.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+The following table assumes all tests will be renamed according to the naming convention proposed in [https://github.com/kyma-project/community/blob/master/collaboration/sig-core/proposals/test-folder-consolidation.md]
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
The following table assumes all tests will be renamed according to the naming convention proposed in [https://github.com/kyma-project/community/blob/master/collaboration/sig-core/proposals/test-folder-consolidation.md].
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+### Tests
+| path | old image name | new image name |
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
| Path | Old image name | New image name |
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+| tests/event-bus/ | event-bus | event-bus-tests |
+| tests/integration/dex/ | dex-integration-tests | dex-integration-tests |
+| tests/knative-build/ | knative-build-acceptance-tests | knative-build-tests |
+| tests/knative-serving/ | knative-serving-acceptance-tests | knative-serving-tests |
+| tests/service-catalog/ | service-catalog-acceptance-tests | service-catalog-tests |
+
+
+## Optional
+
+- automatically generate the image name in the Makefile.
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
- Automatically generate the image name in the Makefile.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+| tests/end-to-end/kubeless/ | kubeless-integration | kubeless-end-to-end-tests |
+| tests/end-to-end/upgrade/ | e2e-upgrade-test | upgrade-end-to-end-tests |
+| tests/event-bus/ | event-bus | event-bus-tests |
+| tests/integration/dex/ | dex-integration-tests | dex-integration-tests |
+| tests/knative-build/ | knative-build-acceptance-tests | knative-build-tests |
+| tests/knative-serving/ | knative-serving-acceptance-tests | knative-serving-tests |
+| tests/service-catalog/ | service-catalog-acceptance-tests | service-catalog-tests |
+
+
+## Optional
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
## Optional actions
```
Did you mean optional actions?

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+
+| path | image name |
+| ---- | ---------- |
+| `tests/end-to-end/upgrade` | `upgrade-end-to-end-test` |
+| `tests/knative-serving` | `knative-serving-test` |
+
+
+## Actions
+
+- Update the developer's guide to include the naming convention
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
- Update the developer's guide to include the naming convention.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+- Update all Makefiles to use the new naming convention
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
- Update all Makefiles to use the new naming convention.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+```shell
+component-name
+└- cmd
+   ├- executable_a
+   |  └- main.go
+   └- executable_b
+      └- main.go
+```
+
+> The `cmd`-folder is not required, but good practise. If a components source code can only be used to create one single binary, it is also ok to put its `main`-function in a file called main.go on the root-level of the components folder
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
> The `cmd` directory is not always required, but it is good practice to use it. If a component's source code can be used to create only a single binary, it is also ok to put its `main` function in a file called `main.go` on the root-level of the `components` folder.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+See the following examples:
+
+| Path | Image name | Description |
+| ---- | ---------- | ---- |
+| components/event-bus/cmd/event-publish-service | event-bus-event-publish-service | Multiple Docker images |
+| components/apiserver-proxy/cmd/proxy | apiserver-proxy | Single Docker image |
+
+### Tests
+
+The naming pattern could be as presented in the following table
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
The naming pattern could be as presented in the following table:
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 15]
```diff
@@ -0,0 +1,100 @@
...
+| tests/event-bus/ | event-bus | event-bus-tests |
+| tests/integration/dex/ | dex-integration-tests | dex-integration-tests |
+| tests/knative-build/ | knative-build-acceptance-tests | knative-build-tests |
+| tests/knative-serving/ | knative-serving-acceptance-tests | knative-serving-tests |
+| tests/service-catalog/ | service-catalog-acceptance-tests | service-catalog-tests |
+
+
+## Optional actions
+
+- Automatically generate the image name in the Makefile.
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]:
```suggestion
- Automatically generate the image name in a Makefile.
```

File: collaboration/sig-core/proposals/image-name-consolidation.md [hunk 16]
```diff
@@ -0,0 +1,100 @@
...
+
+## Problem
+Currently, the correspondence between the image name and the image source code is not always clear.
+
+As a result:
+- It's not easy to find the source code for a particular image.
+- It's not easy to find the latest image that was built using the source code.
+
+## Goal
+
```
Comment (mmitoraj) [inline collaboration/sig-core/proposals/image-name-consolidation.md]: the empty line is not needed

**Pull request merged**